ols/app/main.py

```diff
@@ -53,4 +54,60 @@ async def rest_api_counter(
     return response
 
 
+@app.middleware("")
+async def log_requests_responses(
+    request: Request, call_next: Callable[[Request], Awaitable[Response]]
+) -> Response:
+    """Middleware for logging of HTTP requests and responses, at debug level."""
+    if logger.isEnabledFor(logging.DEBUG):
+        # Log request headers
+        request_log_message = (
+            f"Request {request.client.host}:{request.client.port}"
+            f" {request.headers}, Body: "
+        )
+
+        # Log request body
+        request_body = await request.body()
+        if request_body:
+            request_log_message += f"{request_body.decode('utf-8')}"
+        else:
+            request_log_message += "None"
+
+        logger.debug(request_log_message)
+
+    # Process request
+    response = await call_next(request)
+
+    if logger.isEnabledFor(logging.DEBUG):
+        # Log response headers
+        logger.debug(
+            f"Response {request.client.host}:{request.client.port} {response.headers}"
+        )
+
+        async def stream_response_body(
+            response_body: AsyncGenerator[bytes, None]
+        ) -> AsyncGenerator[bytes, None]:
+            async for chunk in response_body:
+                if logger.isEnabledFor(logging.DEBUG):
+                    logger.debug(
+                        f"Response {request.client.host}:{request.client.port} "
+                        f"chunk: {chunk.decode('utf-8', errors='ignore')}"
+                    )
+                yield chunk
+
+        if logger.isEnabledFor(logging.DEBUG):
```

Comment:
it will be `True` here everytime, right? so no condition needed

Reply:
true, my oversight